api/test/robot_tests/plugin_manager.robot

```diff
@@ -2,20 +2,215 @@
 Documentation       This is a test suite for Thyra Plugin Manager endpoints.
 
 Library             RequestsLibrary
+Library             String
 Resource            variables.resource
+Resource            keywords.resource
+
+Suite Setup         Suite Setup
 
 
 *** Test Cases ***
-GET /plugin-manager
+GET /plugin-manager with no plugins
     ${response}=    GET    ${API_URL}/plugin-manager
     Status Should Be    ${STATUS_OK}
     ${listLength}=    Get Length    ${response.json()}
     Should Be Equal As Integers    ${listLength}    0
 
-GET /plugin-manager/invalid
+POST /plugin-manager?source={{pluginSource}}
+    ${source}=    Set Variable
+    ...    https://github.com/massalabs/thyra-plugin-hello-world/releases/download/0.0.3/thyra-plugin-hello-world_${OS}-${ARCH}.zip
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager
+    ...    params=source=${source}
+    ...    expected_status=${STATUS_NO_CONTENT}
+    Sleep    1 seconds    # Wait for the plugin to be registered
+
+GET /plugin-manager with one plugin
+    ${response}=    GET    ${API_URL}/plugin-manager
+    Status Should Be    ${STATUS_OK}
+    ${listLength}=    Get Length    ${response.json()}
+    Should Be Equal As Integers    ${listLength}    1
+
```

Comment:
Since the test suite is running in the written order, we can make some "scenarios" like that. We could also make them inside of a single test to make it clearer to test some specific behaviours.